### ISSUE TITLE
Unable to Load Credentials Error

### ISSUE THREAD

**Issue by robbyki** [databricks/spark-redshift#305]
Using the below code with spark 2.0.1 but still getting
`Exception in thread "main" com.amazonaws.AmazonClientException: Unable to load AWS credentials from any provider in the chain` error. 

```
    val conf = new SparkConf()
      .setAppName("RedshiftTest")
    val sc = new SparkContext(conf)

    val sqlContext = new org.apache.spark.sql.SQLContext(sc)

//    val sqlContext = new org.apache.spark.sql.SQLContext(sc)

    val Creds = new ProfileCredentialsProvider()
    //    val client = new AmazonS3Client(Creds)
    val s3AccessKey = Creds.getCredentials().getAWSAccessKeyId
    val s3SecretKey = Creds.getCredentials().getAWSSecretKey

    val hadoopConf = sc.hadoopConfiguration
    hadoopConf.set("fs.s3a.impl", "org.apache.hadoop.fs.s3native.NativeS3FileSystem")
    hadoopConf.set("fs.s3a.awsAccessKeyId", s3AccessKey)
    hadoopConf.set("fs.s3a.awsSecretAccessKey", s3SecretKey)

    val tempS3Dir = "s3a://<bucket>/spark_redshift_out2/"
    val dbhost = "<host>"
    val dbuser = "<user>"
    val dbname = "<db_name>"
    val dbpass = "<pass>"
    val dbport = "<port>"
    val jdbcURL = s"jdbc:redshift://$dbhost:$dbport/$dbname?user=$dbuser&password=$dbpass"

    // Get some data from a Redshift table
    val df: DataFrame = sqlContext.read
      .format("com.databricks.spark.redshift")
      .option("url", jdbcURL)
      .option("dbtable", "dfp_data.country")
      .option("tempdir", tempS3Dir)
      .option("forward_spark_s3_credentials", true)
      .load()
    println(df.show())
```

**Comment by JoshRosen** [comment]
```
    hadoopConf.set("fs.s3a.impl", "org.apache.hadoop.fs.s3native.NativeS3FileSystem")
```

Here, aren't you rebinding `s3a` to `s3n` and are then setting keys for `fs.s3a`? I don't think that those settings will be read by S3N so I think you'll have to set the `fs.s3n` configurations instead.

**Comment by kyprifog** [comment]
I have this issue and I am correctly setting "fs.s3a.impl" to "org.apache.hadoop.fs.s3a.S3AFileSystem"

**Comment by gmanolache** [comment]
same issue as @kyprifog

**Comment by kyprifog** [comment]
@gmanolache just so you are aware I got this stable finally but the spark redshift connectors treatment of s3 is kind of touchy.  I got really unexpected behavior with small variations.  I would recommend using the most endorsed authentication methods, tinker with it until it works and then leave it alone.  It's been too long since I've done it but if I get a chance later today I can fill you in on what worked for me
